### DISCUSSION TITLE
Using Intellij with trading app

### DISCUSSION THREAD

**Discussion by westernsam** [gvolpe/trading#23] [General]
Hi,

Thanks for creating this book and project. I really enjoyed pfps and I'm excited to dig around trading app.

I prefer using intellij idea. Unfortunately due to a [long standing issue](https://youtrack.jetbrains.com/issue/SCL-8396) with intellij's dependency model it can't generate a project from the trading app sbt definition. The issue is that intellij doesn't support `compile->test` in a multi-project configuration. 

I have worked round this by moving `generators` to `src` and making `scalacheck` a compile rather than test library. Is there another approach that might be cleaner? 

I think quite a lot of developers use scala with intellij so it would be good if trading app worked in intellij without modification.

Thanks again,
Sam

**Comment by gvolpe** [comment]
Hi @westernsam ,

Thanks for starting the discussion!

I believe that setting is required only by `feed`? If so, you can remove it from the [aggregation](https://github.com/gvolpe/trading/blob/main/build.sbt#L59) to work around the issue. 

However, have you tried using IntelliJ with BSP support? I hear it's working great right now and this should be supported. My fellow Igal wrote a blog post about it sometime ago: https://hmemcpy.com/2021/09/bsp-and-intellij/ 

Let me know how that goes,
Cheers.

**Reply by westernsam** [reply]
Hi,

BSP works perfectly out of the box, so definitely the way to go. Thanks for the response and suggestion.

Cheers,
Sam
